File: backend/src/main/java/org/example/financemanager/dto/GroupedReportDto.java
package org.example.financemanager.dto;

import lombok.Builder;
import lombok.Data;

import java.util.Map;

@Data
@Builder
public class GroupedReportDto {
    private AmountWithCurrencyResponse amount;
    private Map<String, AmountWithCurrencyResponse> dataWithTotalAmountOfCategory;
    private Map<String, Double> dataWithPercentage;
}

File: frontend/src/app/common/grouped-report-dto.ts
import {AmountWithCurrencyResponse} from "./amount-with-currency-response";

export class GroupedReportDto {
  constructor(amount: AmountWithCurrencyResponse, dataWithTotalAmountOfCategory: {
    [p: string]: AmountWithCurrencyResponse
  }, dataWithPercentage: { [p: string]: number }) {
    this.amount = amount;
    this.dataWithTotalAmountOfCategory = dataWithTotalAmountOfCategory;
    this.dataWithPercentage = dataWithPercentage;
  }
  public amount: AmountWithCurrencyResponse;
  public dataWithTotalAmountOfCategory: { [key: string]: AmountWithCurrencyResponse }
  public dataWithPercentage: { [key: string]: number }
}
